Scripts/RemediationScripts/Remediate-SetClassicFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1020 @@
+            $validFrontDoorDetails = $frontdoorDetails | Where-Object { ![String]::IsNullOrWhiteSpace($_.ResourceId) }
+            
+            $validFrontDoorDetails | ForEach-Object {
+                $resourceId = $_.ResourceId
+
+                try
+                {
+                    $frontDoorResource = Get-AzFrontDoor -ErrorAction SilentlyContinue
+                    $FrontDoorResources += $frontDoorResource
+                }
+                catch
+                {
+                    Write-Host "Fetching Front Door resource: Resource ID: [$($resourceId)]. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+                    Write-Host "Skipping this Front Door resource..." -ForegroundColor $([Constants]::MessageType.Warning)
+                }
+            }
+        }
+    }
+    $totalFrontDoors = ($FrontDoorResources | Measure-Object).Count
+
+    if ($totalFrontDoors -eq 0)
+    {
+        Write-Host "No Front Door found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        return
+    }
+  
+    Write-Host "Found [$($totalFrontDoors)] Front Door(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetMinTLSVersionForFrontDoors"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+    
+    # Includes required version of TLS
+    $requiredMinTLSVersion = 1.2
+
+    # Includes Front Doors where Minimum required TLS version is not set to 1.2.
+    $frontDoorsEndpointWithoutReqMinTLSVersion = @()
+
+    # Includes Front Doors that were skipped during remediation. There were errors remediating them.
+    $frontDoorsSkipped = @()
+
+    $isMinTLSVersionSetOnCustomDomain = @()
+
+    $FrontDoorEndpoint = @()
+
+    $frontDoorWithoutReqMinTLSVersion = @()
+
+    Write-Host "[Step 3 of 4] Fetch all Front Door configurations"
+    Write-Host $([Constants]::SingleDashLine)
+
+    $FrontDoorResources | ForEach-Object {
+        $frontDoorResource = $_
+        $resourceId = $_.Id
+        $resourceName = $_.Name
+        #$frontendPoints = $_.FrontendEndPoints
+        $resourceGroupName= $_.Id.Split('/')[4]; 
+        try
+        {
+            Write-Host "Fetching Front Door Endpoint configuration: Resource ID: [$($resourceId)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            
+            $FrontDoorEndpoint = Get-AzFrontDoorFrontendEndpoint -FrontDoorName $resourceName -ResourceGroupName $resourceGroupName
```

Comment:
If we are collection in response, let's use plural for variable names.

Reply:
Fixed